packages/builder/builder-rspack-provider/src/plugins/react.ts

```diff
@@ -16,6 +16,12 @@ export const builderPluginReact = (): BuilderPlugin => ({
         // https://swc.rs/docs/configuration/compilation#jsctransformreactruntime
         runtime: 'automatic',
       });
+
+      setConfig(rspackConfig, 'builtins.provide', {
```

Comment:
Why do we need this new config for Rspack

Reply:
due to this change:
https://github.com/web-infra-dev/rspack/pull/3731/files#diff-6195dbe0219b3b7ba39567cf6de6d629fb3a7b78c582c6b0b2957d1425105922